test/unittest/layers/layers_golden_tests.cpp

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file layer_golden_tests.cpp
+ * @date 09 Sept 2021
+ * @brief Common golden test for nntrainer layers (Param Tests)
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#include <layers_common_tests.h>
+
+#include <fstream>
+#include <type_traits>
+
+#include <base_properties.h>
+#include <layer_context.h>
+#include <layer_devel.h>
+#include <nntrainer_test_util.h>
+#include <util_func.h>
+#include <var_grad.h>
+#include <weight.h>
+
+#pragma GCC diagnostic ignored "-Wunused-local-typedefs"
+
+using namespace nntrainer;
+
+using TensorPacks = std::tuple<
+  std::vector<Weight> /**< weight */, std::vector<Var_Grad> /**< in */,
+  std::vector<Var_Grad> /**< out */, std::vector<Var_Grad> /**< tensors */>;
+
+/**
+ * @brief read Tensor with size check
+ *
+ * @param t tensor to read
+ * @param file file stream
+ */
+static void sizeCheckedReadTensor(Tensor &t, std::ifstream &file,
+                                  const std::string &error_msg = "") {
+  unsigned int sz = 0;
+  checkedRead(file, (char *)&sz, sizeof(unsigned));
+  NNTR_THROW_IF(t.getDim().getDataLen() != sz, std::invalid_argument)
+    << "[ReadFail] dimension does not match at " << error_msg << " sz: " << sz
+    << " dimsize: " << t.getDim().getDataLen() << '\n';
+  t.read(file);
```

Comment:
consider storing the size of each tensor in the file before the tensor. This will provide stronger checks for now.
If you are more enthusiastic, you can even consider flatbuffer :)

Reply:
Yes, size of tensor is stored before the tensor in the layer golden test.
Actually, flatbuffer is my next step along this works :) Thanks for the suggestion!

Reply:
great.